tests/asdf/test_core.py

```diff
@@ -0,0 +1,145 @@
+"""Tests asdf implementations of core module."""
+
+import asdf
+import numpy as np
+import pandas as pd
+import xarray as xr
+
+import weldx.transformations as tf
+from weldx.asdf.extension import WeldxExtension, WeldxAsdfExtension
+
+
+# xarray.DataArray ---------------------------------------------------------------------
+
+
+def get_xarray_example_data_array():
+    """Get an xarray.DataArray for test purposes."""
+    data = np.array([[0, 1], [2, 3]])
+
+    d1 = np.array([-1, 1])
+    d2 = pd.DatetimeIndex(["2020-05-01", "2020-05-03"])
+    coords = {"d1": d1, "d2": d2}
+
+    dax = xr.DataArray(data=data, dims=["d1", "d2"], coords=coords)
+    return dax
+
+
+def test_xarray_data_array_save():
+    """Test if an xarray.DataArray can be written to an asdf file."""
+    dax = get_xarray_example_data_array()
+    tree = {"dax": dax}
+    with asdf.AsdfFile(tree, extensions=[WeldxExtension(), WeldxAsdfExtension()]) as f:
+        f.write_to("xarray.asdf")
+
+
+def test_xarray_data_array_load():
+    """Test if an xarray.DataArray can be restored from an asdf file."""
+    f = asdf.open("xarray.asdf", extensions=[WeldxExtension(), WeldxAsdfExtension()])
+    dax_file = f.tree["dax"]
+    dax_exp = get_xarray_example_data_array()
+    assert dax_exp.identical(dax_file)
+
+
+# xarray.DataArray ---------------------------------------------------------------------
+
+
+def get_xarray_example_dataset():
+    """Get an xarray.Dataset for test purposes."""
+
+    temp = [
+        [[15.0, 16.0, 17.0], [18.0, 19.0, 20.0]],
+        [[21.0, 22.0, 23.0], [24.0, 25.0, 26.0]],
+    ]
+    precip = [[[1.0, 2.0, 3.0], [4.0, 5.0, 6.0]], [[7.0, 8.0, 9.0], [10.0, 11.0, 12.0]]]
+    lon = [[-99.83, -99.32], [-99.79, -99.23]]
+    lat = [[42.25, 42.21], [42.63, 42.59]]
+
+    dsx = xr.Dataset(
+        {
+            "temperature": (["x", "y", "time"], temp),
+            "precipitation": (["x", "y", "time"], precip),
+        },
+        coords={
+            "lon": (["x", "y"], lon),
+            "lat": (["x", "y"], lat),
+            "time": pd.date_range("2014-09-06", periods=3),
+            "reference_time": pd.Timestamp("2014-09-05"),
+        },
+    )
+
+    return dsx
```

Comment:
maybe expand the example with `attrs` dictionary on the Dataset as well as individual DataArrays (simple comments)

Also string coordinates would be interesting here (any other special dtypes we need to look out for?)

Reply:
Added a simple dictionary as attributes in the tests. We still have to consider the string coordinates.